.github/workflows/mdchecks.yml

```diff
@@ -0,0 +1,79 @@
+name: Markdown Checks
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    paths: '**.md'
+
+jobs:
+  mdchecks:
+    name: Markdown Checks
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Checkout the Repo 
+      uses: actions/checkout@v2
+
+    - name: Set up NodeJS
+      uses: actions/setup-node@v1
+
+    - name: Get MD files in this PR
+      uses: dorny/paths-filter@v2
+      id: filter
+      with:
+        # Paths to files will be available in `filter_files` output variable.
+        list-files: shell
+        filters: |
+          markdown:
+            - added|modified: '*.md'
+
+      # This step filters the list of files further, removing from consideration
+      # any file that includes the XML comment
+      # `<!-- BSSW_GITHUB_CI_IGNORE_MD_CHECKS -->` beginning on a first column
+      # anywhere in the file. The resulting list of files is available in
+      # `keep_files`.
+    - name: Skip Ignore Files
+      id: keep
+      run: |
+        mdfiles=""
+        for f in ${{ steps.filter.outputs.markdown_files }}; do 
+          if [ -n "$(grep '^<!-- BSSW_GITHUB_CI_IGNORE_MD_CHECKS -->' $f)" ]; then
+            continue
+          fi
+          mdfiles="$mdfiles $f"
+        done
+        echo ::set-output name=keep_files::${mdfiles}
+
+    - name: Install MD Tools
+      if: ${{ steps.keep.outputs.keep_files != '' }}
+      run: |
+        npm install -g markdownlint-cli
+        npm install -g markdown-spellcheck
+        npm install -g markdown-link-check
+
+      # See https://www.npmjs.com/package/markdownlint
+      # Global lint rules can be set in .markdownlint.json at top of repo.
+    - name: Lint check
+      if: ${{ steps.keep.outputs.keep_files != '' }}
+      run: markdownlint ${{ steps.keep.outputs.keep_files }}
+
+      # See https://www.npmjs.com/package/markdown-link-check
+    - name: Link check
+      if: ${{ steps.keep.outputs.keep_files != '' }}
+      run: markdown-link-check ${{ steps.keep.outputs.keep_files }}
+
+      # See https://www.npmjs.com/package/markdown-spellcheck
+      # Any MD file can have file-specific words specified in an XML comment
+      # block beginning with <!-- BSSW_GITHUB_CI_SPELLING_WORDS. Words are
+      # taken, one per line, from such a comment block and stop with the
+      # terminating XML comment tag -->. A global list of spelling words is
+      # also taken from `.spelling` file a top of repo.
+    - name: Spell check
+      if: ${{ steps.keep.outputs.keep_files != '' }}
+      run: |
+        cp .spelling .spelling.orig
+        for f in ${{ steps.keep.outputs.keep_files }}; do 
+          sed -n '/^<\!-- BSSW_GITHUB_CI_SPELLING_WORDS/,/^-->/p' $f | grep -v -e '--' > .spelling
+          cat .spelling.orig >> .spelling
+          mdspell -a -n -r -x --en-us $f || exit 1
+        done
```

Comment:
Seems like this should be put into a local script called something like `check_markdown_files.sh` and alled something like:

```
      if: ${{ steps.keep.outputs.keep_files != '' }}
      run: |
       <base-dir>/check_markdown_files.sh ${{ steps.keep.outputs.keep_files }}
```

That way, you can develop, debug, and run the script `check_markdown_files.sh` locally.  Win, win, win.

Seems like a best practice would be to minimize what you directly write in a GHA `*.yml` file to be just what is needed to communicate with GH.

Reply:
@markcmiller86, I just ran the raw tool `mdspell` locally on the file `WhenNotToWriteAutomatedTests.md` and it was a fairly nice interactive experience where it created the file `.spelling` for the words I asked it to ignore for this file.   It would be nice to have a wrapper script that will generate the starting file `.spelling` given the contents pulled out of the block `<\!-- BSSW_GITHUB_CI_SPELLING_WORDS ... -->` for a given file and then run the spell check.  Otherwise, the process to fix the errors locally is going to be pretty painful.

Reply:
I am wondering if a more workable workflow would be to require file-specific spelling ignore words for the file `<base-name>.md` to be put into a file called `<base-name>.spelling`.  Then, when a spell check is done locally that generates/updates `.spelling`, you just copy the contents of `.spelling` to `<base-name>.spelling` rather than having to copy them into the block `<\!-- BSSW_GITHUB_CI_SPELLING_WORDS ... -->`.  But it would be nice to keep the spelling ignores in the same file being checked (which programs like `ispell` and `aspell` do) so I am torn.  Either way, having to mess around with creating and moving a `.spelling` file is not a very nice workflow.